File: backend/models/teacher.model.js
/**
 * Teacher document schema (Firestore collection: 'teachers')
 *
 * @typedef {Object} Teacher
 * @property {string} id - Firestore document ID
 * @property {string} name - Full name
 * @property {string} email - Email address
 * @property {string} employeeId - Employee ID
 * @property {string[]} subjects - Subjects taught
 * @property {string[]} classes - Classes assigned
 * @property {string} qualification - Educational qualification
 * @property {string} joinDate - ISO date string
 * @property {'active'|'inactive'} status - Employment status
 * @property {string} createdAt - ISO timestamp
 * @property {string} updatedAt - ISO timestamp
 */

export function validateTeacher(data) {
  const required = ['name', 'email', 'employeeId'];
  const missing = required.filter((field) => !data[field]);
  if (missing.length > 0) {
    throw new Error(`Missing required fields: ${missing.join(', ')}`);
  }
  return true;
}

File: backend/models/teacher_model.py
"""
Teacher document schema (Firestore collection: 'teachers')

Fields:
  id            - Firestore document ID
  name          - Full name
  email         - Email address
  employeeId    - Employee ID
  subjects      - List of subjects taught
  classes       - List of classes assigned
  qualification - Educational qualification
  joinDate      - ISO date string
  status        - Employment status: 'active' | 'inactive'
  createdAt     - ISO timestamp
  updatedAt     - ISO timestamp
"""


def validate_teacher(data):
    required = ['name', 'email', 'employeeId']
    missing = [field for field in required if not data.get(field)]
    if missing:
        raise ValueError(f"Missing required fields: {', '.join(missing)}")
    return True
